### DISCUSSION TITLE
Accessing morph relationship in edit form

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#4735] [Help] [answered]
I have a morph relationship between an `Enquiry` and a `ClientEnquiry` model:

```
//Enquiry
public function enquirable()
{
    return $this->morphTo();
}

//ClientEnquiry
public function enquiry()
{
    return $this->morphTo(Enquiry::class);
}
```
In Filament Admin, I can access the `ClientEnquiry` data in the `Enquiry` table using:
```
TextColumn::make('enquirable.message')
```
However, when it comes to the form...
```
TextInput::make('enquirable.message')
```
...just results in an empty field. I can neither see the data nor save via the field.

I thought that amending `getEloquentQuery()` would do the trick but no joy:
```
public static function getEloquentQuery(): Builder
{
    return parent::getEloquentQuery()->with('enquirable');
}
```
Any idea as to what I'm missing?

**Comment by danharrin** [comment, marked as answer]
Something like this? https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships

Dot-syntax for form fields only handles nested array data, not relationships

**Reply by titantwentyone** [reply]
Thank you, Dan!

**Comment by titantwentyone** [comment]
Just going to add a note here as it is related and may help someone (and future me). When testing forms, you should provide an array for the data of the relationship.

```
   $enquiry = Enquiry::factory()->asClientEnquiry()->create();
   $new_enquiry = Enquiry::factory()->asClientEnquiry()->make();

   livewire(EditEnquiry::class, [
      'record' => $record->getKey()
   ])
   ->fillForm([
      'email' => $new_enquiry->email,
      'enquirable' => [
         'message' => $new_enquiry->enquirable->message
      ])
   ->call('save')
   ->... //make whatever assertions you need.
```
